dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -163,8 +170,28 @@ public static void main(String[] arguments)
         DependencyGraphBuilder.getStaticLinkageCheckDependencies(artifacts);
     List<DependencyPath> dependencyPaths = dependencyGraph.list();
 
+    // To remove duplicates on (groupId:artifactId) for dependency mediation
+    Map<String, String> keyToFirstArtifactVersion = Maps.newHashMap();
```

Comment:
This all seems like it would be better to be done while creating the dependency graph, not afterwards. Could `getStaticLinkageCheckDependencies()` do this, and just include the path to the correct dependency?

Reply:
I prefer this function (the caller of getStaticLinkageCheckDependencies) to pickup correct artifacts.

- `getStaticLinkageCheckDependencies` is responsible to build a graph of dependency.
- The caller of the function is responsible to pick up artifacts (with version) depending on a dependency mediation strategy.
- As of now the strategy is Maven's dependency mediation (pick closest)
- In future we'll add an option to check Gradle's dependency mediation (pick highest version) #309.


Renamed `getStaticLinkageCheckDependencies` to `getStaticLinkageCheckDependencyGraph` to clarify that it returns a graph.

Reply:
side note, but You probably don't need "StaticLinkageCheck" in either of these method names.

Reply:
DependencyGraphBuilder serves other use cases than StaticLinkageCheck. So I'd keep the function names.